matrix/lambdas/daemons/worker.py

```diff
@@ -1,21 +1,89 @@
+import os
+
+import zarr
+
 from matrix.common.lambda_handler import LambdaHandler
+from matrix.common.lambda_handler import LambdaName
+from matrix.common.dss_zarr_store import DSSZarrStore
 from matrix.common.dynamo_handler import DynamoHandler
+from matrix.common.dynamo_handler import DynamoTable
+from matrix.common.dynamo_handler import StateTableField
+from matrix.common.s3_zarr_store import S3ZarrStore
+from matrix.common.pandas_utils import convert_dss_zarr_root_to_subset_pandas_dfs
 
 
 class Worker:
     """
     The worker (third) task in a distributed filter merge job.
     """
-    def __init__(self):
+    def __init__(self, request_id):
         self.lambda_handler = LambdaHandler()
         self.dynamo_handler = DynamoHandler()
+        self._request_id = request_id
+        self._deployment_stage = os.environ['DEPLOYMENT_STAGE']
+        self._bundle_uuid = None
+        self._bundle_version = None
+        self._input_start_row = None
+        self._input_end_row = None
+        self._num_rows = None
+
+    def run(self, format: str, worker_chunk_spec: dict):
+        """Process and write one chunk of dss bundle matrix to s3 and
+        invoke reducer lambda when last worker job is completed.
 
-    def run(self, request_id: str, filter_string: str, worker_chunk_spec: dict):
+        Inputs:
+        format: (str) Expected file format of request
+        worker_chunk_spec: (dict) Information about input bundle and matrix row indices to process
         """
-        Filter one work chunk and invoke reducer lambda when last worker job is completed.
+        self._parse_worker_chunk_spec(worker_chunk_spec)
```

Comment:
I'm wondering if we're doing this if we should just pass in the parameters flat.

Reply:
Leave as is for now but good to note for future improvement.

Reply:
Leavig a comment in code to track this